js/store.js

```diff
@@ -157,6 +306,37 @@ const store = {
     }
   },
 
+  // Checks if third party is on whitelist for first party,
+  // and if the whitelisted third party should be unwhitelisted.
+  whiteListCheck(origin, target, firstParty, thirdParty) {
+    let newThirdParty = false;
+    if (!('isVisible' in thirdParty) || thirdParty['isVisible']) {
+      // third party needs a whitelist check
+      if (this.onWhiteList(origin, target) && !('isVisible' in thirdParty)) {
+        // third party is whitelisted for this first party
+        // ignore the link for now
+        thirdParty['isVisible'] = false;
+        thirdParty['whiteListedFirstParty'] = origin;
```

Comment:
Please just make `firstPartyHostnames` consistent for `isVisible` or not and when it becomes visible loop through `firstPartyHostnames` to clear up to add it to `thirdPartyHostnames` as the website might exist to multiple whitelisted first parties which currently would be a bug.